lib/jekyll/document.rb

```diff
@@ -433,8 +434,8 @@ def respond_to?(method, include_private = false)
     # Override of method_missing to check in @data for the key.
     def method_missing(method, *args, &blck)
       if data.key?(method.to_s)
-        Jekyll.logger.warn "Deprecation:", "Document##{method} is now a key in the #data hash."
-        Jekyll.logger.warn "", "Called by #{caller.first}."
+        Jekyll::Deprecator.deprecation_message "Document##{method} is now a key "\
+                           "in the #data hash. Called by #{caller.first}."
```

Comment:
These should be printed on separate lines. Could you please put "Called by" with its own call to `deprecation_message`?